cobigen-templates/templates-devon4j/src/main/templates/crud_java_server_app/templates/java/${variables.rootPackage}/${variables.component}/common/api/${variables.entityName}.java.ftl

```diff
@@ -3,7 +3,13 @@ package ${variables.rootPackage}.${variables.component}.common.api;
 
 import ${variables.rootPackage}.general.common.api.ApplicationEntity;
 
-public interface ${variables.entityName} extends ApplicationEntity {
+<#assign compositeIdTypeVar = JavaUtil.getReturnTypeOfMethodAnnotatedWith(classObject,"javax.persistence.EmbeddedId")>
+<#if compositeIdTypeVar!="null"> 
+import ${variables.rootPackage}.general.common.api.ApplicationComposedKeyEntity;
+import ${variables.rootPackage}.${variables.component}.common.api.${compositeIdTypeVar};
+</#if>
+
+public interface ${variables.entityName} extends <#if compositeIdTypeVar != "null">ApplicationComposedKeyEntity<${compositeIdTypeVar}><#else>ApplicationEntity</#if> {
```

Comment:
In case of a composite key, we will not generate any inheritance, this has to be done by the user.

Reply:
Changes done and committed. Re-Review required.